### PR TITLE
Improve speed of func version command

### PR DESCRIPTION
**Pull request by ejizba**
### Issue describing the changes in this PR

This has been a pet peeve of mine for a while: the `func --version` command often takes several seconds on my machine. It can be random, but for example it took 18 seconds when I tested it just now. In particular, it annoys me because this slows down the VS Code UI which will check the version of core tools installed for various things.

With my changes, I'm consistently seeing `func --version` take <1 second.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by mattchenderson — APPROVED** [review]
Noting the side effect of dropping the telemetry event. I'm ok with that here. This also bypasses the first time experience, which again, I'm good with. This seems like a great set of changes.

File: src/Azure.Functions.Cli/Program.cs [hunk 1]
```diff
@@ -4,26 +4,39 @@
...
+                _container = InitializeAutofacContainer();
+                AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
+
+                Console.CancelKeyPress += (s, e) =>
+                {
+                    _container.Resolve<IProcessManager>()?.KillChildProcesses();
+                };
+
+                ConsoleApp.Run<Program>(args, _container);
+            }
```
Comment (liliankasem) [inline src/Azure.Functions.Cli/Program.cs:39]: nit: return and leave the rest as it is outside of an if-else brace

```suggestion
                Environment.Exit(ExitCodes.Success);
                return;
            }

            FirstTimeCliExperience();
            SetupGlobalExceptionHandler();
            SetCoreToolsEnvironmentVariables(args);
            _container = InitializeAutofacContainer();
            AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;

            Console.CancelKeyPress += (s, e) =>
            {
                _container.Resolve<IProcessManager>()?.KillChildProcesses();
            };

            ConsoleApp.Run<Program>(args, _container);
```

**Review by liliankasem — DISMISSED** [review]
I think there are plans to revamp this to work a lot more like the dotnet cli, but this should be okay for now!

**Review by liliankasem — DISMISSED** [review]

**Review by liliankasem — APPROVED** [review]

**Pull request merged**